tests/undernames.test.ts

```diff
@@ -0,0 +1,185 @@
+      contract = warp.pst(srcContractId).connect(nonContractOwner);
+      nonContractOwnerAddress = await arweave.wallets.getAddress(
+        nonContractOwner,
+      );
+    });
+    describe('Submits undername increase', () => {
+      describe('with bad input', () => {
+        it.each([
+          '',
+          '*&*##$%#',
+          '-leading',
+          'this-is-a-looong-name-a-verrrryyyyy-loooooong-name-that-is-too-long',
+          'test.subdomain.name',
+          false,
+          true,
+          0,
+          1,
+          3.5,
+        ])(
+          'should throw an error when an invalid name is submitted: %s',
+          async (badName) => {
+            const undernameInput = {
+              name: badName,
+              qty: 1,
+            };
+            const writeInteraction = await contract.writeInteraction(
+              {
+                function: 'increaseUndernames',
+                ...undernameInput,
+              },
+              {
+                disableBundling: true,
+              },
+            );
+
+            expect(writeInteraction?.originalTxId).not.toBe(undefined);
+            const { cachedValue } = await contract.readState();
+            expect(Object.keys(cachedValue.errorMessages)).toContain(
+              writeInteraction!.originalTxId,
+            );
+            expect(
+              cachedValue.errorMessages[writeInteraction!.originalTxId],
+            ).toEqual(expect.stringContaining(INVALID_INPUT_MESSAGE));
+          },
+        );
+
+        it.each([
+          '',
+          '*&*##$%#',
+          '-leading',
+          'this-is-a-looong-name-a-verrrryyyyy-loooooong-name-that-is-too-long',
+          'test.subdomain.name',
+          false,
+          true,
+          0.5,
+          0,
+          Infinity,
+          -Infinity,
+          -1,
+          -1000,
+        ])(
+          'should throw an error when an invalid quantity is provided: %s',
+          async (badQty) => {
+            const undernameInput = {
+              name: 'apple',
+              qty: badQty,
+            };
+            const writeInteraction = await contract.writeInteraction(
+              {
+                function: 'increaseUndernames',
+                ...undernameInput,
+              },
+              {
+                disableBundling: true,
+              },
+            );
+
+            expect(writeInteraction?.originalTxId).not.toBe(undefined);
+            const { cachedValue } = await contract.readState();
+            expect(Object.keys(cachedValue.errorMessages)).toContain(
+              writeInteraction!.originalTxId,
+            );
+            expect(
+              cachedValue.errorMessages[writeInteraction!.originalTxId],
+            ).toEqual(expect.stringContaining(INVALID_INPUT_MESSAGE));
+          },
+        );
+      });
+
+      describe('with valid input', () => {
+        it.each([
+          1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 100, 1000, 10000, 100000, 1000000,
```

Comment:
what's are max allowed undernames?

Reply:
will add permutation calculation for available undername spaces

Reply:
This is a good question - what is the maximum amount of undernames we want to support for a given ANT? 

Ill start a slack thread on this.

Reply:
added